NICE.Identity.Management/ClientApp/src/views/AddOrganisation/AddOrganisation.tsx

```diff
@@ -0,0 +1,233 @@
+import React, { Component } from "react";
+import { Link } from "react-router-dom";
+import { Alert } from "@nice-digital/nds-alert";
+import { Breadcrumbs, Breadcrumb } from "@nice-digital/nds-breadcrumbs";
+import { Button } from "@nice-digital/nds-button";
+import { Grid, GridItem } from "@nice-digital/nds-grid";
+import { PageHeader } from "@nice-digital/nds-page-header";
+import { Input } from "@nice-digital/nds-input";
+import { Endpoints } from "../../data/endpoints";
+import { fetchData } from "../../helpers/fetchData";
+import { isDataError } from "../../helpers/isDataError";
+import { ErrorMessage } from "../../components/ErrorMessage/ErrorMessage";
+
+type CustomError = {
+	error: Error;
+	dataMessage: string;
+};
+
+type AddOrganisationState = {
+	formName: string;
+	validationError: boolean;
+	validationErrorMessage: string;
+	orgNameBlockedArray: Array<string>;
+	hasSubmitted: boolean;
+	error?: Error;
+	isSaveButtonLoading: boolean;
+};
+export class AddOrganisation extends Component<
+	Record<string, never>,
+	AddOrganisationState
+> {
+	constructor(props = {}) {
+		super(props);
+
+		this.state = {
+			formName: "",
+			validationError: false,
+			validationErrorMessage:
+				"Organisation name should be alphanumeric and be between 2-100 characters",
+			orgNameBlockedArray: [],
+			hasSubmitted: false,
+			isSaveButtonLoading: false,
+		};
+
+		document.title = "NICE Accounts - Add organisation";
+	}
+
+	toggleValidationMessage = (blockedOrgNameFound: boolean): string => {
+		return blockedOrgNameFound
+			? "Organisation already exists - name should be unique"
+			: "Organisation name should be alphanumeric and be between 2-100 characters";
+	};
+
+	handleSubmit = async (
+		e: React.FormEvent<HTMLFormElement>,
+	): Promise<void | boolean> => {
+		e.preventDefault();
+		this.setState({ isSaveButtonLoading: true });
+
+		const form = e.currentTarget;
+		const blockedOrgNameFound = this.state.orgNameBlockedArray.includes(
+			this.state.formName,
+		);
+		let hasSubmitted = true;
+		let formName = "";
+
+		if (!form.checkValidity() || blockedOrgNameFound) {
+			this.setState({ validationError: true, isSaveButtonLoading: false });
+			return false;
+		}
+
+		const fetchOptions = {
+			method: "POST",
+			headers: { "Content-Type": "application/json" },
+			body: JSON.stringify({
+				name: this.state.formName,
+			}),
+		};
+
+		const organisation = await fetchData(
+			Endpoints.organisationsList,
+			fetchOptions,
+			true,
+		);
+
+		if (isDataError(organisation)) {
+			const errorObject = organisation as CustomError;
+
+			if (
+				errorObject.dataMessage.indexOf("that organisation already exists") > -1
```

Comment:
This feels like a pretty fragile way to check what type of error we've got back.

Reply:
Yeah, I completely agree. I had a similar issue doing this here:

https://github.com/nice-digital/identity-management/blob/2f811322899bddf7a12a6ab32cc57254f0b39146/NICE.Identity.Management/ClientApp/src/views/EditUser/EditUser.tsx#L156-L175

I think I agreed with JW that the error would send an obscure code back, which I could use. I'm not sure what the best approach is for the backend and front end for ascertaining that it's a certain error, but wanted to use the error to show a validation-style inline error message rather than the failed-type error component.